crates/mem-pool/src/sync/mq/gw_kafka.rs

```diff
@@ -0,0 +1,130 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use gw_types::bytes::Bytes;
+use gw_types::packed::{RefreshMemBlockMessage, RefreshMemBlockMessageUnion};
+use gw_types::prelude::{Builder, Entity, Reader};
+use kafka::client::{Compression, FetchOffset, GroupOffsetStorage};
+use kafka::producer::{AsBytes, Record};
+
+use crate::sync::fan_in::FanInMemBlock;
+
+use super::{Consume, Produce};
+
+pub(crate) struct Producer {
+    producer: kafka::producer::Producer,
+    topic: String,
+}
+
+impl Producer {
+    pub(crate) fn new(hosts: Vec<String>, topic: String) -> Result<Self> {
+        let mut client = kafka::client::KafkaClient::new(hosts);
+        client.load_metadata(&[&topic]).map_err(|err| {
+            anyhow!(
+                "topic {:?} not found, load metadata error: {:?}",
+                &topic,
+                err
+            )
+        })?;
+        let producer = kafka::producer::Producer::from_client(client)
+            .with_ack_timeout(Duration::from_secs(1))
```

Comment:
1 second? I'm not sure if it works under pressure.

Reply:
I will put this in config. This needs to be fine-tuned.